hypha/apply/users/views.py

```diff
@@ -39,6 +43,21 @@ class AccountView(UpdateView):
     def get_object(self):
         return self.request.user
 
+    def form_valid(self, form):
+        updated_email = form.cleaned_data['email']
+        name = form.cleaned_data['full_name']
+        slack = form.cleaned_data['slack']
+        user = get_object_or_404(User, id=self.request.user.id)
+        if updated_email and updated_email != user.email:
+            base_url = reverse('users:confirm_password')
+            query_string = urlencode({
+                'updated_email': updated_email,
+                'name': name,
+                'slack': slack
+            })
+            return redirect('{}?{}'.format(base_url, query_string))
```

Comment:
@sandeepsajan0, let's not put these variables in url, that would be introducing another security issue while fixing one..rather this could be hidden fields in the password form if possible.

**OR** we could use [Django's cryptographic signing](https://docs.djangoproject.com/en/3.2/topics/signing/), this also allows us to fix the second point in our list..session time out one...e.g. by using [TimestampSigner](https://docs.djangoproject.com/en/3.2/topics/signing/#verifying-timestamped-values) we could know how much time user took to fill up the password and raise error accordingly.

Also, we need to make sure this PasswordForm(related url/view) is only getting accessed on redirect from this view. User should not be able to access it in any other way. We could maybe make the signing parameter in the url mandatory.